grails-app/services/com/unifina/service/StorageNodeService.groovy

```diff
@@ -12,25 +12,53 @@ class StorageNodeService {
 	StreamrClientService streamrClientService
 
 	List<Stream> findStreamsByStorageNode(EthereumAddress storageNodeAddress) {
-		List<StreamStorageNode> items = StreamStorageNode.findAllByStorageNodeAddress(storageNodeAddress.toString())
-		Iterable<Serializable> streamIds = items.collect { it.streamId } as Iterable<Serializable>
-		return Stream.getAll(streamIds)
+		List<StreamStorageNode> nodes = StreamStorageNode.findAllByStorageNodeAddress(storageNodeAddress.toString())
+		List<Stream> results = new ArrayList<>()
+		for (StreamStorageNode node : nodes) {
+			Stream stream = node.getStream()
+			results.add(stream)
+		}
+		return results
 	}
 
-	List<StreamStorageNode> findStorageNodesByStream(String streamId) {
-		boolean streamExists = (Stream.get(streamId) != null)
-		if (streamExists) {
-			return StreamStorageNode.findAllByStreamId(streamId)
+	Set<StreamStorageNode> findStorageNodesByStream(String streamId) {
+		Stream stream = streamService.getStream(streamId)
+		if (stream != null) {
+			return stream.getStorageNodes()
 		} else {
 			throw new NotFoundException("Stream", streamId)
 		}
 	}
 
+	StreamStorageNode findStorageNodeByAddress(Stream stream, EthereumAddress address) {
+		for (final StreamStorageNode node : stream.getStorageNodes()) {
+			if (node.getStorageNodeAddress() == address.toString()) {
+				return node
+			}
+		}
+		return null
+	}
+
+	boolean hasStorageNodeWithAddress(Stream stream, EthereumAddress address) {
```

Comment:
This method could be simplified to `findStorageNodeByAddress(s, a) != null` (or could remove the method)

Reply:
I've got patch coming up after this one that removes the method. It's a fix for BACK-155